etc/deltaDebug.py

```diff
@@ -219,11 +220,6 @@ def perform_step(self, cut_index=-1):
             odb.write_def(self.base_db.getChip().getBlock(),
                           self.base_def_file)
 
-        # Destroy the DB in memory to avoid being out-of-memory when
-        # the step code is running
-        if (self.base_db is not None):
-            self.base_db.destroy(self.base_db)
-
```

Comment:
Why is this removed?

Reply:
Because the database instance is being used later in the code, that’s why there was a segmentation fault

Reply:
Once the step is over it shouldn't be needed.  Where is it used later?  Why wasn't this an issue before?

Reply:
get_cuts function gets called after perform_step, get_cuts calls get_elms which calls get_insts and get_nets and both of these functions access the database instance.
I do not know why it was not an issue before, however, from the CI, the deltaDebug test has never passed.

Reply:
@oharboe it looks like a bug you introduced in https://github.com/The-OpenROAD-Project/OpenROAD/commit/c18ba0d6172d9741acb4e9a4adabbd3503d423a2 where you added the `while j < self.get_cuts():` loop

Reply:
I don't think removing the call to destroy is the correct fix.  Probably it is better to gather whatever stats we need before the db is destroyed.

Reply:
Let me know if there is anything I can do to help. I dont think the python interface should ever segfault, even if there is a python error, right?

A helpful error message as the nature of the problem rather, from which it should be a small matter to fix the python.

The python code now has unit and integration tests, but did @vvbandeira hook them up yet?

Reply:
Habiba is doing that now and hit this issue.  Accessing objects from a destroyed db is equivalent to access deallocated memory.